tests/integration_tests/test_with_master.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+from asyncio import AbstractEventLoop
+from pathlib import Path
+from typing import List, cast
+
+import mock
+import pytest
+import pytest_asyncio
+from mock import create_autospec
+
+from tickit.core.components.component import Component, ComponentConfig
+from tickit.core.components.device_simulation import DeviceSimulation
+from tickit.core.management.event_router import InverseWiring
+from tickit.core.management.schedulers.master import MasterScheduler
+from tickit.core.runner import run_all_forever
+from tickit.core.state_interfaces.state_interface import get_interface
+from tickit.core.typedefs import ComponentException, ComponentID, PortID
+from tickit.devices.sink import SinkDevice
+from tickit.utils.configuration.loading import read_configs
+
+
+@pytest.fixture
+def configs_from_yaml() -> List[ComponentConfig]:
+    path_to_yaml = Path(__file__).parent / "configs" / "test_with_master.yaml"
+    return read_configs(path_to_yaml)
+
+
+@pytest.fixture
+def wiring(configs_from_yaml: List[ComponentConfig]) -> InverseWiring:
+    return InverseWiring.from_component_configs(configs_from_yaml)
+
+
+@pytest.fixture
+def components(configs_from_yaml) -> List[Component]:
+    return [config() for config in configs_from_yaml]
+
+
+@pytest_asyncio.fixture
+async def master_scheduler(
+    wiring: InverseWiring,
+    components: List[Component],
+    event_loop: AbstractEventLoop,
+):
+    with mock.patch(
+        "tickit.devices.sink.SinkDevice.update",
+        return_value=create_autospec(SinkDevice.update),
+    ):
+        scheduler = MasterScheduler(wiring, *get_interface("internal"))
+
+        assert scheduler.running.is_set() is False
+        run_task = event_loop.create_task(
+            run_all_forever(
```

Comment:
nit: please may you change the use of run_all_forever? We're looking at removing it in another change. (There is no way you'd have known this so no worried)